rest-service/manager_rest/test/endpoints/test_app.py

```diff
@@ -29,3 +32,75 @@ def test_get_root_404(self):
         """
         resp = self.app.get('/')
         self.assertEqual(404, resp.status_code)
+
+    def test_does_not_init_tracer(self):
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+        self.jaeger_mock_config.assert_not_called()
+
+
+@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
+class AppWithTracingNoIPTestCase(base_test.BaseServerTestCase):
+    """Test the basic setup of the tracer and dispatch wrapping for the
+    tracing when tracing is enabled but no endpoint IP is set.
+    """
+
+    def setUp(self):
+        super(AppWithTracingNoIPTestCase, self).setUp()
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+
+    def create_configuration(self):
```

Comment:
it is repeating itself needs to go to a public file or in the base test case class

Reply:
How would I create different setups then? I don't think there's a different option here.